docs/docs/specs/v0.2/composer.md

```diff
@@ -195,10 +213,115 @@ Other plugins possible:
 3. the **github** plugin - which provides nodes representing github issues and assets
 
 ## Example Plugins
+
 See `/packages/experimental/surface`.
 
-## Things to think about:
+## Graph
+
+This is a contract for UI affordances which present, organize and navigate over the graph of user knowledge.
+
+The goal is to enable any kind of navigation UI such as trees, lists, accordions of lists, accordions of trees, force-directed layouts, ..., etc. and to allow them to source their content/navigation structure from plugins.
+
+The average tree or list item has mostly similar properties:
+
+```ts
+type Node = {
+  label: string;
+  icon: any;
+  // ... 
+  children: Node[]; // if a tree
+}
+```
+
+The `graph` contract is derived by working backwards from these UI models which are chosen to view and navigate "everything".
+
+- users use a TreeView to organize and navigate items in ECHO spaces
+- users can select nodes in the TreeView
+- selected tree nodes are presented in detail on the right side of the app via `<Surface />`
+  - if multiple nodes are selected, the detail area shows only the total number of things selected
```

Comment:
multi-select could render as a stack and give you the option to save that stack? could initially require them to all be in the same space but this could be an on-ramp for cross-space stacks